new smart contracts/contracts/Collection.sol

```diff
@@ -3,36 +3,91 @@
-import {FragnovaBaseUri} from "./FragnovaBaseUri.sol";
+import {Initializable} from "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import {OwnableUpgradeable} from "./OwnableUpgradeable.sol";
 import {UpdatableOperatorFilterer} from "operator-filter-registry/src/UpdatableOperatorFilterer.sol";
+import {MerkleRootNFT} from "./MerkleRootNFT.sol";
 import {ERC2981} from "@openzeppelin/contracts/token/common/ERC2981.sol";
-import "hardhat/console.sol";
+import {IOperatorFilterRegistry} from "operator-filter-registry/src/IOperatorFilterRegistry.sol";
+import {ERC721A} from "erc721a/contracts/ERC721A.sol";
+
+/* Graph of inheritance (Note: "NFT" is short for "MerkleRootNFT")
 
-address constant OPENSEA_REGISTRY = address(
-    0x000000000000AAeB6D7670E522A718067333cd4E
-);
-address constant DEFAULT_OPENSEA_SUBSCRIPTION = address(
-    0x3cc6CddA760b79bAfa08dF41ECFA224f810dCeB6
-);
+OwnableUpgradeable    UpdatableOperatorFilterer
+            \     \  /      /
+             \    NFT     /
+              \    |    /
+               Collection
 
+*/
+/// @title A Fragnova Collection
+/// @author Fragcolor Pte. Ltd.
+/// @notice To sell a Fragnova Collection Smart Contract on OpenSea, the owner of the contract must register the contract on the OpenSea Registry (https://etherscan.io/address/0x000000000000AAeB6D7670E522A718067333cd4E#code) by either calling `register()`, `registerAndSubscribe()` or `registerAndCopyEntries()` .
+/// @notice To stop selling on OpenSea, the owner of the contract must unregister the contract on the OpenSea Registry by calling `unregister()`.
+/// @notice The OperatorFilter checks can be bypassed if you calls the function `updateOperatorFilterRegistryAddress()`.
 contract Collection is
-    Ownable,
-    ERC721A("", ""),
-    UpdatableOperatorFilterer(
-        OPENSEA_REGISTRY,
-        DEFAULT_OPENSEA_SUBSCRIPTION,
-        true
-    ),
+    Initializable,
+    OwnableUpgradeable,
+    UpdatableOperatorFilterer,
+    MerkleRootNFT,
     ERC2981
 {
+    /// @notice OpenSea's Operator Filter Registry (https://github.com/ProjectOpenSea/operator-filter-registry#deployments)
+    address constant OPENSEA_FILTER_REGISTRY =
+        address(0x000000000000AAeB6D7670E522A718067333cd4E);
+    /// @notice OpenSea's Curated Subscription Address (https://github.com/ProjectOpenSea/operator-filter-registry#deployments)
+    address constant DEFAULT_OPENSEA_SUBSCRIPTION =
+        address(0x3cc6CddA760b79bAfa08dF41ECFA224f810dCeB6);
     /// @notice Price of minting an element in the collection
     uint256 public mintPrice;
 
-    /// @dev This is non-`constant` only for testing purposes
-    FragnovaBaseUri public baseUriProxy;
+    /// @notice This should never run
+    constructor()
+        UpdatableOperatorFilterer(
+            OPENSEA_FILTER_REGISTRY,
+            DEFAULT_OPENSEA_SUBSCRIPTION,
+            true
+        )
+        ERC721A("", "")
+    {}
```

Comment:
put an assert here if should never run?

Reply:
Actually it will run when we deploy the first `ProtoCollection`/`InstanceCollection` contracts which will be used as a state variable in the `CollectionFactory` contract